tests/ui/loop-match/loop-match.rs

```diff
@@ -0,0 +1,44 @@
+// Test a basic correct example of #[loop_match] with #[const_continue].
+
+//@ run-pass
+
+#![allow(incomplete_features)]
+#![feature(loop_match)]
+
+enum State {
+    A,
+    B,
+    C,
+}
+
+fn main() {
+    let mut state = State::A;
+    #[loop_match]
+    'a: loop {
+        state = 'blk: {
+            match state {
+                State::A => {
+                    #[const_continue]
+                    break 'blk State::B;
+                }
+                State::B => {
+                    // without special logic, the compiler believes this is a re-assignment to
+                    // an immutable variable because of the `loop`. So, this tests that local
+                    // variables work.
```

Comment:
```suggestion
                    // Without special logic, the compiler believes this is a
                    // reassignment to an immutable variable because of the
                    // `loop`. So this tests that local variables work.
```